### PR TITLE
[Docs] PR for Need istioCompatability flag when creating k8s platform config (Doc9293)

### PR DESCRIPTION
**Pull request by lizayugabyte**
1. Added the istio compatibility info.
2. Edited the whole file to ensure clarity.

### CODE REVIEW

**Comment by lizayugabyte** [comment]
@akumar-yb Could you, please, review this change so we'd be able to close the PR?

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md [hunk 1]
```diff
@@ -95,63 +95,56 @@ Run the following `kubectl` command to apply the YAML file:
...
-The following output should appear:
+Expect the following output:
 
 ```
 serviceaccount/yugabyte-platform-universe-management created
 ```
 
-The next step is to grant access to this ServiceAccount using ClusterRoles/Roles and ClusterRoelBindings/RoleBindings, thus allowing it to manage the YugabyteDB universe's resources for you.
-Follow any one of the following steps depending on your requirements.
+You need to grant access to this ServiceAccount using ClusterRoles and Roles as well as ClusterRoelBindings and RoleBindings, thus allowing it to manage the YugabyteDB universe's resources for you.
```
Comment (ddhodge) [inline docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md:104]: ClusterRoleBindings?

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md [hunk 2]
```diff
@@ -175,130 +168,122 @@ You can create a `kubeconfig` file for previously created `yugabyte-platform-uni
...
-
-- **Kube Config** is *optional* if specified at provider level or else `required`
-
-- **Namespace**, *optional* if provided SA have the `Cluster Admin` permissions else `required`. The SA used in provided `kubeconfig` should have access to this namespace.
+- Use the `Region` field to select the region.
+- Use the **Zone** field to select a zone label that should match with your failure domain zone label `failure-domain.beta.kubernetes.io/zone`.
+- Optionally, use the **Storage Class** field to enter a comma-delimited value. If you do not specify this value, it would default to standard. You need to ensure that this storage class exists in your Kubernetes cluster.
+- Use the **Namespace** field to specify the namespace. If provided SA has the `Cluster Admin` permissions, you are not required to complete this field. The SA used in the provided `kubeconfig` file should have access to this namespace.
+- Use **Kube Config** to upload the configuration file. If this file is available at provider level, you are not required to supply it.
```
Comment (ddhodge) [inline docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md:206]: First time 'SA' used, just spell out? eg - If the provided ServiceAccount
The ServiceAccount used

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md [hunk 3]
```diff
@@ -175,130 +168,122 @@ You can create a `kubeconfig` file for previously created `yugabyte-platform-uni
...
+      annotation2: 'bar'
+    ```
 
-```yml
-networkAnnotation:
-  annotation1: 'foo'
-  annotation2: 'bar'
-```
+  - Overrides to add custom resource allocation for YB master and TServer pods and it overrides the instance types selected in the Yugabyte universe creation flow:
+
```
Comment (ddhodge) [inline docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md:256]: Overrides to add custom resource allocation for YB master and TServer pods and override the instance types selected in the Yugabyte universe creation flow

**Review by ddhodge — APPROVED** [review]
A few comments

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md (hunk 1 above)
Comment (lizayugabyte) [reply docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md:104]: yes!

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md (hunk 2 above)
Comment (lizayugabyte) [reply docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md:206]: Changed to ServiceAccount

File: docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md (hunk 3 above)
Comment (lizayugabyte) [reply docs/content/latest/yugabyte-platform/configure-yugabyte-platform/set-up-cloud-provider/kubernetes.md:256]: I think it should be:
Overrides to add custom resource allocation for YB master and TServer pods (this overrides the instance types selected in the Yugabyte universe creation flow):

**Review by akumar-yb — APPROVED** [review]
Looks good to me.

**Pull request merged**